### PR TITLE
dnsdist: Add regression tests for DNS header set/get via Lua FFI

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by omoerbeek — COMMENTED** [review]
I find it hard to follow the logic of these tests.

The `testLuaFFISetAAHeader` claims to test the `AA=0` case, but REFUSED is returned for the `aa` case in `setAAResponseAction`, if I'm not mistaken.

Also but maybe that's related to me not knowing the API very well): `DNSResponseAction.HeaderModify` is returned while the header is not modified and DNSResponseAction.None` is returned when the header *is* modified.

**Comment by rgacogne** [comment]
> I find it hard to follow the logic of these tests.
> 
> The `testLuaFFISetAAHeader` claims to test the `AA=0` case, but REFUSED is returned for the `aa` case in `setAAResponseAction`, if I'm not mistaken.

Both tests use the same `setAAResponseAction` Lua function which checks whether the AA bit is clear in the initial response, and:
- set it if it was clear, as tested by `testLuaFFISetAAHeader`
- change the rcode to `REFUSED` if it was set, at tested by `testLuaFFIgetAAHeader`

> Also but maybe that's related to me not knowing the API very well): `DNSResponseAction.HeaderModify` is returned while the header is not modified and `DNSResponseAction.None` is returned when the header _is_ modified.

You are right that the two return values are not very consistent. It does not really matter for these cases because the modification has already been applied in both cases and there is no other rules, so `DNSResponseAction.None` would work, but the intent of returning `DNSResponseAction.HeaderModify`  is to prevent subsequent rules from being executed once we have decided to return a `REFUSED` response.

I'll refactor the tests and add some comments.

**Comment by rgacogne** [comment]
I updated the tests, splitting the function in two and adding comments.

**Review by omoerbeek — APPROVED** [review]
